docs/drivers/azure.md

```diff
@@ -31,15 +31,27 @@ authenticate:
     Microsoft Azure: To sign in, use a web browser to open the page https://aka.ms/devicelogin.
     Enter the code [...] to authenticate.
 
-After authenticating, the driver will remember your credentials up to two weeks.
+- Browser based authentication **credentials expire in 2 weeks**.
+- Once your credentials expire, you will be prompted to log in again.
+- This is practical for daily use; but not recommended for headless applications.
 
-> **KNOWN ISSUE:** There is a known issue with Azure Active Directory causing stored
-> credentials to expire within hours rather than 14 days when the user logs in with
-> personal Microsoft Account (formerly _Live ID_) instead of an Active Directory account.
-> Currently, there is no ETA for resolution, however in the meanwhile you can
-> [create an AAD account][aad-docs] and login with that as a workaround.
+## Authentication (service principal account)
 
-[aad-docs]: https://azure.microsoft.com/documentation/articles/virtual-machines-windows-create-aad-work-id/
+1. Create an Azure Service Principal account ([described here][sp]) to get a
+   `client_id` and `client_secret`
+2. Use `azure role assignment create` command to give your service account
+   `Owner` or `Contributor` roles on your subscription (see article above).
+3. Pass `--azure-client-id` and `--azure-client-secret` arguments (or
+   environment variables) to to `docker-machine`.
+
+Azure Service Principal Account credentials are recommended for headless
+applications (such as CI/CD systems).
+
+The Service Principal Account password created through the `azure ad app create`
```

Comment:
@ppadala there is a linked article above in this section. (see `[described here]`). 

I believe the AAD service accounts are free. You can learn more here: https://azure.microsoft.com/en-us/pricing/details/active-directory/ It is the de facto way of providing long-lived creds to any app consuming Azure API and it lets you do RBACs etc.